### DISCUSSION TITLE
balance/balance

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#207] [General]
# balance/balance

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/balance/balance.html

**Comment by dsencheng** [comment]
手机浏览，底部的下页链接总是没有，现在又多了公众号广告，还没关闭按钮

**Reply by isno** [reply]
移动端的公众号广告关掉了。书籍是用 Vuepress 写的，不是完整的电子书系统。觉得内容可以，回来等纸质的出来，买纸质的吧

**Comment by 8Haixin** [comment]
在看到这篇文章之前，一直认为每个请求-响应周期都会建立一个连接。

**Reply by joy717** [reply]
HTTP 1.0 是这样的，每次请求，响应结束后就会关闭连接。频繁建立连接，导致性能有损耗。
HTTP 1.1 以及后面HTTP2，再往后的HTTP3（QUIC）都不会立刻关闭连接，在一定时间内再次请求，会复用原有的连接.
但复用连接引入一个“队头阻塞”的问题，不过这个就是另外的一个问题了.

**Comment by punnpkin** [comment]
原来四层负载是这个意思

**Comment by zjyyzhengli** [comment]
四层负载均衡器模型下，client建立TCP连接时的目的IP地址是LB的IP吗？
图4-2上LB和backend之间的TCP连接是什么意思？对应的IP和端口信息如何理解呢？

**Reply by isno** [reply]
你继续看看 4.4 节，完整介绍了 四层负载均衡技术。

**Comment by Zhe-MingC** [comment]
这里七层负载均衡器应该是4.1.2不是6.1.2
